### PR TITLE
fix: convert collection to list before randomizing it

### PR DESCRIPTION
**Pull request by witsch**
`plone.batching` returns a `BaseBatch` object, which causes `Population must be a sequence.  For dicts or sets, use sorted(d).` when passed to `random.sample`.

### CODE REVIEW

**Comment by witsch** [comment]
Oops, I just saw @alecpm already fixed this with #36. Should have checked before starting to debug I guess… sorry about that. I'm not closing the PR for now as we're using it in our buildout, but please feel free to do so once Alec's fix has been merged… 😊👋

**Comment by davisagli** [comment]
@witsch I merged Alec's PR -- but I see that that one was for the master branch, and yours is for 3.3.x

**Comment by witsch** [comment]
> @witsch I merged Alec's PR -- but I see that that one was for the master branch, and yours is for 3.3.x

Good catch! 😊 Might be easier to merge it than to backport 'master' then…

However, I could also change it to "mimic" Alec's version first. To keep the differences as small as possible I mean. In any case, we're still on Plone 5 so it'd be nice to have it in the 3.3 branch as well.

File: news/randomization.fix
Comment (davisagli) [inline news/randomization.fix:1]: @witsch This file should be renamed to 42.bugfix to match the pattern that towncrier looks for.

**Review by davisagli — COMMENTED** [review]
@witsch I think it's fine either way. I don't want to spend more of my time on a fix for Plone 5.2, but both fixes make sense to me, and I think it would be fine to merge this one if the jenkins builds are green. But note this comment about the file name for the changelog entry...

**Comment by davisagli** [comment]
@jenkins-plone-org please run jobs

File: news/randomization.fix
Comment (mauritsvanrees) [reply news/randomization.fix:1]: I have renamed it.

**Review by mauritsvanrees — APPROVED** [review]
LGTM.
There was one random robot failure, and several plone.app.caching failures that I am had already fixed on master and am back porting in https://github.com/plone/plone.app.caching/pull/135.
We can merge the current PR after merging my caching one.

**Pull request merged**
